improver_tests/set_up_test_cubes.py

```diff
@@ -60,48 +62,131 @@ def construct_xy_coords(ypoints, xpoints, spatial_grid):
+            provided, a grid is created centred around (0,0).
 
     Returns:
         y_coord, x_coord (tuple):
             Tuple of iris.coords.DimCoord instances
     """
+    if spatial_grid not in GRID_COORD_ATTRIBUTES.keys():
+        raise ValueError("Grid type {} not recognised".format(spatial_grid))
+
+    if grid_spacing is None:
+        if domain_corner is None:
+            y_array, x_array = _default_grid(ypoints, xpoints, spatial_grid)
+        else:
+            raise ValueError("Grid spacing required to setup grid from domain corner.")
+    else:
+        if domain_corner is None:
+            domain_corner = _set_domain_corner(ypoints, xpoints, grid_spacing)
+        y_array, x_array = _create_y_x_arrays(
+            ypoints, xpoints, domain_corner, grid_spacing
+        )
+
+    y_coord = DimCoord(
+        y_array,
+        GRID_COORD_ATTRIBUTES[spatial_grid]["yname"],
+        units=GRID_COORD_ATTRIBUTES[spatial_grid]["units"],
+        coord_system=GRID_COORD_ATTRIBUTES[spatial_grid]["coord_system"],
+    )
+    x_coord = DimCoord(
+        x_array,
+        GRID_COORD_ATTRIBUTES[spatial_grid]["xname"],
+        units=GRID_COORD_ATTRIBUTES[spatial_grid]["units"],
+        coord_system=GRID_COORD_ATTRIBUTES[spatial_grid]["coord_system"],
+    )
+
+    return y_coord, x_coord
+
+
+def _create_y_x_arrays(ypoints, xpoints, domain_corner, grid_spacing):
+    """
+    Creates arrays for constructing y and x DimCoords.
+
+    Args:
+        ypoints (int):
+            Number of grid points required along the y-axis
+        xpoints (int):
+            Number of grid points required along the x-axis
+        domain_corner (list of int or float):
+            Bottom left corner of grid domain [y,x] (degrees or metres)
+        grid_spacing (int or float):
+            Grid resolution (degrees or metres)
+
+    Returns:
+        y_array, x_array (tuple):
+            Tuple of numpy.ndarrays
+    """
+    y_points_array = [domain_corner[0] + i * grid_spacing for i in range(ypoints)]
+    x_points_array = [domain_corner[1] + i * grid_spacing for i in range(xpoints)]
+    y_array = np.array(y_points_array, dtype=np.float32)
+    x_array = np.array(x_points_array, dtype=np.float32)
+
+    return y_array, x_array
+
+
+def _set_domain_corner(ypoints, xpoints, grid_spacing):
+    """
+    Set domain corner when none provided.
+
+    Args:
+        ypoints (int):
+            Number of grid points required along the y-axis
+        xpoints (int):
+            Number of grid points required along the x-axis
+        grid_spacing (int or float):
+            Grid resolution (degrees or metres).
+
+    Returns:
+        domain_corner (list of int or float):
+            [y,x] values of the bottom left corner of the domain
+    """
+    y_start = 0 - ((ypoints - 1) * grid_spacing) / 2
+    x_start = 0 - ((xpoints - 1) * grid_spacing) / 2
+    domain_corner = [y_start, x_start]
+
+    return domain_corner
```

Comment:
```suggestion
    return y_start, x_start
```
A tuple return may be better here since the output will always be of length 2.

Reply:
Modified in commit 6d2f80ab51f7b5adc4806f9efdf99631cd469ef7.

Changed all instances of `domain_corner` to tuple for consistency as isn't, and shouldn't need to be, changed at all.